addons/website/models/website_edited.py

```diff
@@ -171,3 +171,31 @@ def _find_url(self, url, is_default_local_url=False):
                 f"'{url}?",
             ])
         return self._find(likes)
+
+    @api.model
+    def _clean(self):
+        for res_model, res_field, _ in self._get_website_edited_html_fields():
+            removed_models = []
+            if res_model in removed_models:
+                continue
+            if res_model not in self.env:
+                removed_models.append(res_model)
+                self.env.cr.execute(SQL("""
+                    DELETE FROM website_edited
+                    WHERE res_model = {res_model}
+                """).format(
+                    res_model=Placeholder('res_model'),
+                ), {
+                    'res_model': res_model,
+                })
+            elif res_field not in self.env[res_model]:
+                self.env.cr.execute(SQL("""
+                    DELETE FROM website_edited
+                    WHERE res_model = {res_model}
+                    AND res_field = {res_field}
+                """).format(
+                    res_model=Placeholder('res_model'),
+                    res_field=Placeholder('res_field'),
+                ), {
+                    'res_field': res_field,
+                })
```

Comment:
I guess not using the ORM to do a `search([]).unlink()` is really more fast but.. is it really?
I don't expect millions of records in this new table, isn't this premature optimization?

Well, if we can turn this double query into a single one, it would already look more acceptable IMO, with the list of fields retrieved with an util somewhere (I can help if you don't find anything let me know)